src/frontend/src/planner/select.rs

```diff
@@ -98,9 +99,9 @@ impl Planner {
         // Plan the SELECT clause.
         // TODO: select-agg, group-by, having can also contain subquery exprs.
         let has_agg_call = select_items.iter().any(|expr| expr.has_agg_call());
-        if !group_by.is_empty() || having.is_some() || has_agg_call {
+        if !group_by.is_empty() || !grouping_sets.is_empty() || having.is_some() || has_agg_call {
             (root, select_items, having) =
-                LogicalAgg::create(select_items, group_by, having, root)?;
+                LogicalAgg::create(select_items, group_by, grouping_sets, having, root)?;
```

Comment:
Is either `group_by` non-empty or `grouping_set` non-empty? If so, what about making it an enum in `LogicalAggBuilder` and `BoundSelect`?

Reply:
Yes, that is true for `LogicalAggBuilder` and `BoundSelect`.